### PR TITLE
[doc] A few doc cleanups

### PR DESCRIPTION
**Pull request by msfschaffner**

### CODE REVIEW

File: hw/ip/clkmgr/doc/theory_of_operation.md [hunk 1]
```diff
@@ -213,6 +213,7 @@ The life cycle controller only requests the io clock input to be switched.
 When the life cycle controller requests external clock, a request signal `lc_clk_byp_req_i` is sent from `lc_ctrl` to `clkmgr`.
 `clkmgr` then forwards the request to `ast` through `io_clk_byp_req_o`, which performs the actual clock switch and is acknowledged through `io_clk_byp_ack_i`.
 When the clock switch is complete, the clock dividers are stepped down by a factor of 2 and the life cycle controller is acknowledged through `lc_clk_byp_ack_o`.
+E.g. if the internal clock has nominal a frequency of 96MHz, the external clock is expected to have a frequency of only 48MHz after the clock switch is complete.
```
Comment (msfschaffner) [inline hw/ip/clkmgr/doc/theory_of_operation.md:216]: @dkabely-ot @zi-v FYI

[the remark this answers is not in the document]
Comment (msfschaffner) [reply hw/ip/clkmgr/doc/theory_of_operation.md]: thanks!

File: hw/ip/clkmgr/doc/theory_of_operation.md [hunk 2]
```diff
@@ -213,6 +213,7 @@ The life cycle controller only requests the io clock input to be switched.
 When the life cycle controller requests external clock, a request signal `lc_clk_byp_req_i` is sent from `lc_ctrl` to `clkmgr`.
 `clkmgr` then forwards the request to `ast` through `io_clk_byp_req_o`, which performs the actual clock switch and is acknowledged through `io_clk_byp_ack_i`.
 When the clock switch is complete, the clock dividers are stepped down by a factor of 2 and the life cycle controller is acknowledged through `lc_clk_byp_ack_o`.
+E.g. if the internal clock has a nominal frequency of 96MHz, the external clock is expected to have a frequency of only 48MHz after the clock switch is complete.
```
Comment (matutem) [inline hw/ip/clkmgr/doc/theory_of_operation.md:216]: This seems unclear. Perhaps pointing to the frequencies table below will help. The nominal IO clock frequency "is" 96 HMz, and the external clock frequency "must be" 48 MHz. Stepping down the clock dividers causes the indicated frequencies for io_div2 and io_div4.

File: hw/ip/pwrmgr/doc/theory_of_operation.md [hunk 3]
```diff
@@ -145,9 +145,8 @@ There are 4 reset requests in the system
 
 Flash brownout is handled separately and described in [flash handling section](#flash-handling) below.
 
-Peripheral requested resets such as watchdog are handled directly by the power manager, while the non-debug module reset is handled by the reset controller.
-This separation is because the non-debug reset does not affect the life cycle controller, non-volatile storage controllers and alert states.
-There is thus no need to sequence its operation like the others.
+Note that the non-debug module reset is handled similarly to a peripheral requested reset.
+The only differentiation is that the non-debug module reset won't affect the debug module state and associated TAP muxing logic inside the pinmux.
```
Comment (matutem) [inline hw/ip/pwrmgr/doc/theory_of_operation.md:149]:
```suggestion
Note that the non-debug module reset is handled similarly to a peripheral requested reset, except
that the non-debug module reset won't affect the debug module state and associated TAP muxing logic inside the pinmux.
```

**Review by matutem — APPROVED** [review]
I left a couple suggestions for you to sort out :)
LGTM otherwise

File: hw/ip/clkmgr/doc/theory_of_operation.md (hunk 2 above)
Comment (msfschaffner) [reply hw/ip/clkmgr/doc/theory_of_operation.md]: Good idea, let me do that.

File: hw/ip/pwrmgr/doc/theory_of_operation.md (hunk 3 above)
Comment (msfschaffner) [reply hw/ip/pwrmgr/doc/theory_of_operation.md]: thanks, updated!

**Pull request merged**